src/Workspaces/Remote/ServiceHub/Services/SnapshotService.JsonRpcAssetSource.cs

```diff
@@ -63,9 +63,9 @@ public override async Task<bool> IsExperimentEnabledAsync(string experimentName,
                 }
             }
 
-            private bool ReportUnlessCanceled(Exception ex, CancellationToken cancellationToken)
+            private bool ReportUnlessCanceled(Exception ex)
             {
-                if (!cancellationToken.IsCancellationRequested && _owner.IsDisposed)
+                if (!(ex is OperationCanceledException) && !(ex is TaskCanceledException) && _owner.IsDisposed)
```

Comment:
😕 Why the exclusion for `TaskCanceledException`?

Reply:
Because the crash dumps show that the exception is being thrown here and we fail fast. If that's not expected then we'll need to figure out why and fix that.